src/architecture/merkledb.md

```diff
@@ -0,0 +1,207 @@
+# Exonum MerkleDB
+
+Exonum MerkleDB is a persistent storage implementation based on
+[RocksDB][rocks-db]. MerkleDB provides APIs to work with merkelized data
+structures.
+
+MerkleDB is an object database. Objects represent the highest abstraction level
+for the [data storage](storage.md). All objects fall into the following groups:
+
+- **blobs** which represent sequences of bytes
+- **root objects** that do not have parents. These objects have UTF-8
+  identifiers, for example, "block", "state". Root objects can contain blob
+  items inside them.
+
+## Root Object Types
+
+Root objects perform the same role as in relational database
```

Comment:
What role? This paragraph doesn't say that, and I'm not sure if this comparison helps without that.